File: project_3/imp_2/main.js
function maxPrime(n) {
    var maxPrime = -1;
    while (n % 2 == 0) {
        maxPrime = 2;
        n = n / 2;
    }
    for (var i = 3; i <= Math.sqrt(n); i += 2) {
        while (n % i == 0) {
            maxPrime = i;
            n = n / i;
        }
    }
    maxPrime = n > 2 ? n : maxPrime;
    return maxPrime;
}
console.log(maxPrime(600851475143));

File: project_3/imp_2/main.py
import math
  
# A function to find largest prime factor
def maxPrimeFactors (n):
      
    # Initialize the maximum prime factor
    # variable with the lowest one
    maxPrime = -1
      
    # Print the number of 2s that divide n
    while n % 2 == 0:
        maxPrime = 2
        n /=2    # equivalent to n /= 2

          
    # n must be odd at this point, 
    # thus skip the even numbers and 
    # iterate only for odd integers
    number= n
    array = range(3, int(math.sqrt(n)))
    for i in range(3, int(math.sqrt(n)) + 1, 2):

        while number % i == 0:
            print(i, number , number/i)
            maxPrime = i
            number = number / i

      
    # This condition is to handle the 
    # case when n is a prime number 
    # greater than 2
    if number > 2:
        maxPrime = number
      
    return int(maxPrime)
  
# Driver code to test above function
n = 600851475143
print(maxPrimeFactors(n))
